include/pybind11/pybind11.h

```diff
@@ -1979,6 +1979,16 @@ inline std::pair<decltype(internals::registered_types_py)::iterator, bool> all_t
         // gets destroyed:
         weakref((PyObject *) type, cpp_function([type](handle wr) {
             get_internals().registered_types_py.erase(type);
+
+            // Actually just `std::erase_if`, but that's only available in C++20
```

Comment:
The code for the erasure was actually copied and modified from https://github.com/pybind/pybind11/blob/e450eb62c21c8518c1988c3b17bc5793ea347756/include/pybind11/detail/class.h#L218
It would be better to create a helper function as you suggested, but I didn't want to add it in this PR as it isn't isn't connected to the issue. I think it would be better to add it as a separate commit.